### PR TITLE
fix(dispatch): improved handling of undefined simulations in simulations browse view; closes #2999

### LINKED ISSUE #2999: Missing simulations in table view cause 404 page.
The user should only be directed to the 404 page if they try to look at a particular missing resource

The simulation run in the table should not trigger 404s

### PR DESCRIPTION
**Pull request by jonrkarr**
closes #2999

### CODE REVIEW

File: apps/dispatch/src/app/components/simulations/view/view.service.ts [hunk 1]
```diff
@@ -84,14 +87,14 @@ export class ViewService {
...
       id: simulation.id,
-      name: simulation.name,
-      simulator: simulation.simulator,
-      simulatorVersion: simulation.simulatorVersion,
-      cpus: simulation.cpus || 1,
-      memory: simulation.memory || 8,
-      maxTime: simulation.maxTime || 20,
-      envVars: simulation.envVars || [],
-      status: simulation.status,
+      name: simulation?.name as string,
```
Comment (bilalshaikh42) [inline apps/dispatch/src/app/components/simulations/view/view.service.ts:90]: @jonrkarr. This effectively breaks type checking as type assertions override the compiler. I am not sure If this is the best approach to use here. Anything downstream of this method now needs to manually ensure that the data is actually a string and not null or undefined.

File: apps/dispatch/src/app/components/simulations/view/view.component.ts [hunk 2]
```diff
@@ -299,9 +300,15 @@ export class ViewComponent implements OnInit, OnDestroy {
 
   private initSimulationRun(): void {
     this.Simulation$ = this.simulationService
-      .getSimulation(this.uuid)
+      .getSimulation(this.uuid)      
       .pipe(shareReplay(1));
 
+    this.Simulation$.subscribe((simulation: Simulation): void => {
```
Comment (bilalshaikh42) [inline apps/dispatch/src/app/components/simulations/view/view.component.ts:306]: The additional subscription here is redundant with the formatted simulation below. We can just add this to the chain below to ensure that we don't pass undefined into the format simulation method, which would also avoid the type assertion problem

File: apps/dispatch/src/app/components/simulations/view/view.service.ts (hunk 1 above)
Comment (jonrkarr) [reply apps/dispatch/src/app/components/simulations/view/view.service.ts:90]: I realize this. That class is being used for two things. More work is needed to better separate this out.

**Pull request merged**